test/ParseExceptionTest.cpp

```diff
@@ -38,4 +38,13 @@ TEST(ParseException, MetadataGeneration) {
                 (ExceptionMetadata{"where  { ?a a:b ?b }", 12, 14}));
     }
   }
+  // "%" doesn't match any valid token. So in this case we will get an Error
+  // from the Lexer. There is no ExceptionMetadata available for Lexer Errors.
```

Comment:
See comment above.

Reply:
I have two remarks on those tests that I forgot during the initial review:
1. You should assert, that the code in fact throws the correct exception (e.G. add a `FAIL()` statement after the parsing lines.
2. I think you can compare `std::optional` directly with its value type, so you can just write
```
EXPECT_EQ(metadata, someMetadata) // no `.value()` or `.has_value()`
EXPECT_EQ(metadata, std::nullopt);  // see 3.

3. Then (no special treatment of the `nullopt` case (if needed at all), you can make this an abstraction
`expectParseExceptionWithMetadata(std::string_view input, const optional<ExceptionMetaData&> metaData)`